include/dlaf/matrix/matrix.h

```diff
@@ -296,7 +296,10 @@ class Matrix<const T, D> : public internal::MatrixBase {
   }
 
 protected:
-  Matrix(Distribution distribution) : internal::MatrixBase{std::move(distribution)} {}
+  Matrix(Distribution distribution) : internal::MatrixBase{std::move(distribution)} {
+    DLAF_ASSERT((distribution.offset() == GlobalElementIndex{0, 0}), "not supported",
+                distribution.offset());
```

Comment:
This is fine by me, and I think conveys the purpose quite well, but we do have `DLAF_UNIMPLEMENTED` as well. If I understand correctly, this restriction is hopefully temporary so not worth spending time on it, but perhaps in the future some macro-unification/development needs to happen.

Reply:
I had a quick look about how we currently use it, and just provide values of what is not provided. I'm not sure it will be clear for the user what it is not implemented without looking at the code.

```suggestion
   if (dist_matrix_.block_size() != dist_matrix_.tile_size())
     DLAF_UNIMPLEMENTED(dist_matrix_.block_size(), dist_matrix_.tile_size());
```

Would you for an additional message like next?

```suggestion
   if (dist_matrix_.block_size() != dist_matrix_.tile_size())
     DLAF_UNIMPLEMENTED("block_size != tile_size", dist_matrix_.block_size(), dist_matrix_.tile_size());
```